cuda/factorization/ilu_kernels.cu

```diff
@@ -57,7 +56,6 @@ void compute_lu(std::shared_ptr<const DefaultExecutor> exec,
 {
     const auto id = exec->get_device_id();
     auto handle = exec->get_cusparse_handle();
-    gko::cuda::device_guard g{id};
```

Comment:
Do you remember why this was only added for ic and ilu, but not other preconditioners?

Reply:
no, I must have not remembered that Executor already sets them.